common/common-rest/src/main/java/org/apache/servicecomb/common/rest/codec/param/QueryProcessorCreator.java

```diff
@@ -43,6 +54,12 @@ public Object getValue(HttpServletRequest request) throws Exception {
         value = request.getParameterValues(paramPath);
       } else {
         value = request.getParameter(paramPath);
+        if (value == null || value.equals("")) {
```

Comment:
1."" should not change to default value?
2.only process query parameter? what about header/cookie and so on?

Reply:
Maybe we can add defaultValue to base interface

Reply:
"" comes only when user inputs the same and is considered as valid input parameter.

Reply:
OK. For  http://10.57.65.159:8080/codeFirstSpringmvc/reduce?a=
give default Value is better.